skel-usb.c

```diff
@@ -508,4 +513,69 @@ void save_settings(void) {
 
 void set_save_settings(void) {
 	update_calib=1;
-}
\ No newline at end of file
+}
+//Called from the vm, escape hatch to handle messages specific to thymio.
+//Returns true if a message was handled
+int AsebaHandleThymioSpecificMessage(AsebaVMState* vm, uint16_t id, uint16_t* data, uint16_t dataLength) {
+    if(id == ASEBA_MESSAGE_THYMIO_GET_THYMIO_DEVICE_INFO || id == ASEBA_MESSAGE_THYMIO_SET_THYMIO_DEVICE_INFO) {
+        uint8_t* bytes = (uint8_t*)data;
+        uint16_t size  = dataLength * 2;
+        if(size < 1) // We need at least an info type
+            return;
+        uint8_t type = bytes[0];
+        if(type > THYMIO_DEVICE_INFO_ENUM_COUNT) // Send an error ?
+            return;
+
+        if (id == ASEBA_MESSAGE_THYMIO_GET_THYMIO_DEVICE_INFO) {
+            uint8_t size = 0;
```

Comment:
so you have two size variable with two different type and meaning ... no way.

Reply:
Size of the message, and size of the variable we are setting. Although "payload_size" is a poor choice of name, I'll change that.